Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -109,73 +109,74 @@ final class TestDiscoveryCommand: CustomLLBuildCommand, TestBuildCommand {
     }
 
     private func execute(fileSystem: Basics.FileSystem, tool: TestDiscoveryTool) throws {
-        let index = self.context.buildParameters.indexStore
-        let api = try self.context.indexStoreAPI.get()
-        let store = try IndexStore.open(store: TSCAbsolutePath(index), api: api)
-
-        // FIXME: We can speed this up by having one llbuild command per object file.
-        let tests = try store.listTests(in: tool.inputs.map { try TSCAbsolutePath(AbsolutePath(validating: $0.name)) })
-
         let outputs = tool.outputs.compactMap { try? AbsolutePath(validating: $0.name) }
-        let testsByModule = Dictionary(grouping: tests, by: { $0.module.spm_mangledToC99ExtendedIdentifier() })
-
-        func isMainFile(_ path: AbsolutePath) -> Bool {
-            path.basename == TestDiscoveryTool.mainFileName
-        }
 
-        var maybeMainFile: AbsolutePath?
-        // Write one file for each test module.
-        //
-        // We could write everything in one file but that can easily run into type conflicts due
-        // in complex packages with large number of test targets.
-        for file in outputs {
-            if maybeMainFile == nil && isMainFile(file) {
-                maybeMainFile = file
-                continue
+        switch self.context.buildParameters.testingParameters.library {
+        case .swiftTesting:
+            for file in outputs {
+                try fileSystem.writeIfChanged(path: file, string: "")
```

Comment:
are these empty files required / used?

Reply:
Removing them has proven somewhat more difficult than you might think. For the purposes of this experimental feature _right now_, I'm leaving them in, but we will definitely want to revisit authoring them later.